### ISSUE TITLE
CPUID and NFD support: AVX_VNNI

### ISSUE THREAD

**Issue by ionutnechita** [kubernetes-sigs/node-feature-discovery#844]
<no-template>

Hi NFD Team,

I have ADL CPU, but i not have AVX_VNNI in label list.
I have v0.11.0 image and this issue is observed.

```
{
  "beta.kubernetes.io/arch": "amd64",
  "beta.kubernetes.io/os": "linux",
  "feature.node.kubernetes.io/cpu-cpuid.ADX": "true",
  "feature.node.kubernetes.io/cpu-cpuid.AESNI": "true",
  "feature.node.kubernetes.io/cpu-cpuid.AVX": "true",
  "feature.node.kubernetes.io/cpu-cpuid.AVX2": "true",
  "feature.node.kubernetes.io/cpu-cpuid.CETIBT": "true",
  "feature.node.kubernetes.io/cpu-cpuid.CETSS": "true",
  "feature.node.kubernetes.io/cpu-cpuid.CMPXCHG8": "true",
  "feature.node.kubernetes.io/cpu-cpuid.FMA3": "true",
  "feature.node.kubernetes.io/cpu-cpuid.FXSR": "true",
  "feature.node.kubernetes.io/cpu-cpuid.FXSROPT": "true",
  "feature.node.kubernetes.io/cpu-cpuid.IBPB": "true",
  "feature.node.kubernetes.io/cpu-cpuid.LAHF": "true",
  "feature.node.kubernetes.io/cpu-cpuid.MOVBE": "true",
  "feature.node.kubernetes.io/cpu-cpuid.MOVDIR64B": "true",
  "feature.node.kubernetes.io/cpu-cpuid.MOVDIRI": "true",
  "feature.node.kubernetes.io/cpu-cpuid.OSXSAVE": "true",
  "feature.node.kubernetes.io/cpu-cpuid.SCE": "true",
  "feature.node.kubernetes.io/cpu-cpuid.SERIALIZE": "true",
  "feature.node.kubernetes.io/cpu-cpuid.SHA": "true",
  "feature.node.kubernetes.io/cpu-cpuid.STIBP": "true",
  "feature.node.kubernetes.io/cpu-cpuid.VMX": "true",
  "feature.node.kubernetes.io/cpu-cpuid.WAITPKG": "true",
  "feature.node.kubernetes.io/cpu-cpuid.X87": "true",
  "feature.node.kubernetes.io/cpu-cpuid.XSAVE": "true",
  "feature.node.kubernetes.io/cpu-hardware_multithreading": "true",
  "feature.node.kubernetes.io/cpu-model.family": "6",
  "feature.node.kubernetes.io/cpu-model.id": "154",
  "feature.node.kubernetes.io/cpu-model.vendor_id": "Intel",
  "feature.node.kubernetes.io/cpu-pstate.scaling_governor": "powersave",
  "feature.node.kubernetes.io/cpu-pstate.status": "active",
  "feature.node.kubernetes.io/cpu-pstate.turbo": "true",
  "feature.node.kubernetes.io/cpu-rdt.RDTL2CA": "true",
  "feature.node.kubernetes.io/kernel-config.NO_HZ": "true",
  "feature.node.kubernetes.io/kernel-config.NO_HZ_IDLE": "true",
  "feature.node.kubernetes.io/kernel-version.full": "5.15.0-40-generic",
  "feature.node.kubernetes.io/kernel-version.major": "5",
  "feature.node.kubernetes.io/kernel-version.minor": "15",
  "feature.node.kubernetes.io/kernel-version.revision": "0",
  "feature.node.kubernetes.io/pci-0300_8086.present": "true",
  "feature.node.kubernetes.io/pci-0300_8086.sriov.capable": "true",
  "feature.node.kubernetes.io/storage-nonrotationaldisk": "true",
  "feature.node.kubernetes.io/system-os_release.ID": "ubuntu",
  "feature.node.kubernetes.io/system-os_release.VERSION_ID": "22.04",
  "feature.node.kubernetes.io/system-os_release.VERSION_ID.major": "22",
  "feature.node.kubernetes.io/system-os_release.VERSION_ID.minor": "04",
  "gpu.intel.com/cards": "card0",
  "kubernetes.io/arch": "amd64",
  "kubernetes.io/hostname": "10.10.10.10",
  "kubernetes.io/os": "linux",
  "node-role.kubernetes.io/worker": "true"
}

```
```
cpuid -1 | grep AVX-VNNI
      AVX-VNNI: AVX VNNI neural network instrs = true
```

This CPU feature is not implemented in NFD?

BR,
Ionut Nechita
ionutnechita

**Comment by marquiz** [comment]
Hmm yes, we don't support it yet. TBH I wasn't even aware of AVX-VNNI (non AVX512-variant) 😇

On cpuid we rely on https://github.com/klauspost/cpuid so could you submit a feature request there (or even submit a PR implementing the feature 😊)? When we get that in the cpuid package we can pull it in NFD

**Comment by ionutnechita** [comment]
This issue is solved in "klauspost/cpuid#106"

https://github.com/klauspost/cpuid/releases/tag/v2.1.0

It's possible to update NFD version with this fix?

**Comment by mythi** [comment]
> This issue is solved in "[klauspost/cpuid#106](https://github.com/klauspost/cpuid/issues/106)"
> 
> https://github.com/klauspost/cpuid/releases/tag/v2.1.0
> 
> It's possible to update NFD version with this fix?

See #851
